test/framework/src/main/java/org/opensearch/test/ParameterizedOpenSearchIntegTestCase.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.test;
+
+import org.junit.After;
+import org.junit.Before;
+import org.opensearch.action.admin.cluster.settings.ClusterUpdateSettingsResponse;
+import org.opensearch.common.settings.Settings;
+
+/**
+ * Base class for running the tests with parameterization of the dynamic settings
+ * For any class that wants to use parameterization, use @ParametersFactory to generate
+ * different params for dynamic settings. Refer SearchCancellationIT for an example.
+ * Note: this doesn't work for the parameterization of feature flag settings.
+ */
+public abstract class ParameterizedOpenSearchIntegTestCase extends OpenSearchIntegTestCase {
+
+    protected final Settings dynamicSettings;
+
+    public ParameterizedOpenSearchIntegTestCase(Settings dynamicSettings) {
```

Comment:
I don't like how this is just a plain `Settings` type (which can contain any kind of setting!) and the only clue that only "dynamic" settings are supported is the parameter name. However, I don't have any immediate better suggestion, and this is a nice way of parameterizing the concurrent search tests, so I'm okay moving forward with this. Also this a test class we're free to refactor in any way we see fit in the future.

Reply:
For now, I have also added the java doc on the top of the class to state that only the dynamic settings should be used for parameterization. But I agree that we need a strong check to prevent the set of the other static settings in future.

Reply:
Could we add an assertion here to verify the setting type?

Reply:
@dbwiddis thanks for the suggestion, we could do something like this to verify if its an index scoped or node scoped dynamic setting, what do u think?
```
@Before
    public void beforeTests() {
        SettingsModule settingsModule = new SettingsModule(dynamicSettings);
        for (String key : dynamicSettings.keySet()) {
            assertTrue(settingsModule.getClusterSettings().isDynamicSetting(key) || settingsModule.getIndexScopedSettings().isDynamicSetting(key));
        }
        client().admin().cluster().prepareUpdateSettings().setPersistentSettings(dynamicSettings).get();
    }
```